bool state
---
bool success
string message
